packages/format-library/src/link/index.native.js

```diff
@@ -86,13 +85,21 @@ export const link = {
 					let startIndex = value.start;
 					let endIndex = value.end;
 
-					while ( find( value.formats[ startIndex ], startFormat ) ) {
+					while (
+						value.formats[ startIndex ].find(
+							( format ) => format?.type === startFormat.type
+						)
+					) {
 						startIndex--;
 					}
 
 					endIndex++;
 
-					while ( find( value.formats[ endIndex ], startFormat ) ) {
+					while (
+						value.formats[ endIndex ].find(
```

Comment:
Similar to the `startIndex`, this can be `undefined`. 
```suggestion
						value.formats[ endIndex ]?.find(
```

Reply:
Addressed as part of 6e7ec5fe30b2db3aa06859ae115fda878a6f8cea.